### PR TITLE
feat(open-klant): add GetPartijDigitaleAdressenAsync to OpenKlantApiClient

### LINKED ISSUE #445: 🛠 Task: Voeg GetPartijDigitaleAdressenAsync toe aan OpenKlantApiClient
# 🛠 Task: Voeg `GetPartijDigitaleAdressenAsync` toe aan `OpenKlantApiClient`

---

## Intent

De medewerker moet de digitale adressen van een partij kunnen zien wanneer de betrokkene geen eigen adressen heeft. Hiervoor is een methode nodig die de digitale adressen van een partij ophaalt via de Open Klant API.

---

## Functional Description

Voeg een methode `GetPartijDigitaleAdressenAsync(string partijUuid)` toe aan `IOpenKlantApiClient` en de implementatie in `OpenKlantApiClient`. De methode roept `GET /partijen/{uuid}?expand=digitaleAdressen` aan en mapt het response naar een `List<DigitaleAdres>`. Breid het bestaande `Partij`-model uit met een `DigitaleAdressen` property.

---

## Design Reference

N/A — geen design voor deze taak.

---

### In scope

- Nieuwe methode `GetPartijDigitaleAdressenAsync` op `IOpenKlantApiClient`
- Implementatie in `OpenKlantApiClient` met `GET /partijen/{uuid}?expand=digitaleAdressen`
- Uitbreiden `Partij`-model met `List<DigitaleAdres>? DigitaleAdressen` property
- Logging conform bestaande patronen

### Out of scope

- Fallback-logica in `InternetaakDetailsService` (Task 2)
- Frontend wijzigingen
- Foutafhandeling bij onbereikbare API (Task 2 bepaalt strategie)

---

### Request

```
GET /partijen/{uuid}?expand=digitaleAdressen
```

### Response

```json
{
  "uuid": "string",
  "url": "string",
  "naam": "string",
  "_expand": {
    "digitaleAdressen": [
      {
        "uuid": "string",
        "url": "string",
        "adres": "string",
        "soortDigitaalAdres": "email|telefoonnummer",
        "isStandaardAdres": true,
        "omschrijving": "string"
      }
    ]
  }
}
```

### Events / Side Effects

Geen — read-only operatie.

---

## Technical Approach

- Voeg `DigitaleAdressen` property toe aan het bestaande `Partij`-model in `Klantcontact.cs`
- Voeg interface-methode toe aan `IOpenKlantApiClient`
- Implementeer in `OpenKlantApiClient` met `GetFromJsonAsync` + expand response mapping
- Gebruik bestaand `DigitaleAdres`-model (reeds gedefinieerd in `Klantcontact.cs:64`)

### Aggregates & Invariants

- Aggregates: Partij (read-only)
- Invariants: Geen — dit is een read operatie

---

## Impacted Areas

- Backend: `InterneTaakAfhandeling.Common/Services/OpenKlantApi/OpenKlantApiClient.cs`, `Models/Klantcontact.cs`
- Frontend: Geen
- Database: N/A
- APIs: Open Klant API (`/partijen/{uuid}?expand=digitaleAdressen`)
- External systems: Open Klant

---

## Gherkin

```gherkin
Feature: Digitale adressen ophalen van partij

  Scenario: Partij met digitale adressen
    Given een partij met een gekoppeld e-mailadres en telefoonnummer
    When de digitale adressen van de partij worden opgehaald
    Then worden het e-mailadres en telefoonnummer van de partij teruggegeven

  Scenario: Partij zonder digitale adressen
    Given een partij zonder gekoppelde digitale adressen
    When de digitale adressen van de partij worden opgehaald
    Then wordt een lege lijst teruggegeven

  Scenario: Partij niet gevonden
    Given een partij-identifier die niet bestaat in Open Klant
    When de digitale adressen van de partij worden opgehaald
    Then wordt een fout gelogd en een lege lijst teruggegeven
```

---

## Edge Cases & Risks

- Partij UUID verwijst naar een niet-bestaande partij → graceful degradation (log + return empty)
- Open Klant API geeft een expand-response zonder `_expand.digitaleAdressen` veld → behandel als lege lijst

---

## Observability

- Logs: `LogInformation` bij ophalen partij, `LogWarning` bij niet-gevonden of API-fout
- Metrics: Geen nieuwe metrics vereist
- Alerts: Geen

---

## Dependencies

Geen — dit is de eerste taak in de keten.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: Toevoeging van optioneel veld aan model, bestaande code niet geraakt
- Data migration: N/A

---

## Definition of Done

- [ ] Contracts match implementation
- [ ] All Gherkin scenarios covered by tests (Phase 2 E2E)
- [ ] Edge cases handled
- [ ] Observability in place
- [ ] Code reviewed
- [ ] Linting passes (`npm run lint:ci`)
- [ ] Formatting passes (`npm run format:ci`)

---

## Test Mapping

| Gherkin Scenario | Test file | Status |
| ---------------- | --------- | ------ |
| Partij met digitale adressen | Phase 2 E2E | Pending |
| Partij zonder digitale adressen | Phase 2 E2E | Pending |
| Partij niet gevonden | Phase 2 E2E | Pending |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

De medewerker moet de digitale adressen van een partij kunnen zien wanneer de betrokkene geen eigen adressen heeft. Deze task voegt de `GetPartijDigitaleAdressenAsync` methode toe aan de `OpenKlantApiClient` — de eerste bouwsteen voor de partij-fallback logica (Feature #367).

## Changes

- **Unit test infrastructuur**: Nieuw `InterneTaakAfhandeling.Common.Test` project met xUnit, Moq en coverlet — eerste unit test project in het ITA-systeem
- **Model uitbreiding**: `Partij` model uitgebreid met `PartijExpand` class en `_expand` JSON mapping voor `DigitaleAdressen`
- **Interface + implementatie**: `GetPartijDigitaleAdressenAsync(string partijUuid)` op `IOpenKlantApiClient` — doet `GET /partijen/{uuid}?expand=digitaleAdressen` met graceful degradation (lege lijst bij 404/fout)

## Test plan

- [x] Partij met digitale adressen → e-mailadres en telefoonnummer worden teruggegeven
- [x] Partij zonder digitale adressen → lege lijst
- [x] Partij niet gevonden (404) → lege lijst + warning gelogd
- [x] Expand veld ontbreekt in response → lege lijst (edge case)

## Related

Closes #445

### CODE REVIEW

**Review by nijmra — COMMENTED** [review]
## PR Review — #448: feat(open-klant): add GetPartijDigitaleAdressenAsync

**Linked task:** #445 | **Spec coverage:** 3/3 scenarios covered (+1 edge case)

| Dimension | Result |
|-----------|--------|
| Spec adherence | ✅ PASS — all 3 Gherkin scenarios have matching unit tests |
| Contract compliance | ✅ PASS — response shape matches spec (_expand.digitaleAdressen) |
| Impacted Areas | ✅ PASS — changes within declared scope |
| Code quality | ✅ PASS — clean implementation, follows existing patterns |
| Test coverage | ✅ PASS — 4 tests covering happy/empty/404/missing-expand |
| DoD | ✅ PASS — contracts match, edge cases handled, observability present |

**Verdict: APPROVE WITH COMMENTS**

---

### Suggested (non-blocking)

1. **[Test]** The 404 test is named `ReturnsEmptyListAndLogs` but doesn't verify the `LogWarning` call. Either add a Moq `Verify` on the logger or rename the test method.

2. **[Test]** `MockHttpMessageHandler` ignores the request URL — consider capturing the request and asserting that the path is `partijen/{uuid}?expand=digitaleAdressen` to catch endpoint regressions.

3. **[Infra]** The test project has its own `nuget.config` — confirm this is intentional vs. inheriting a solution-level config.

> Note: Suggestions 1 and 2 align with findings from Copilot's automated review.

File: InterneTaakAfhandeling.Common.Test/Services/OpenKlantApi/GetPartijDigitaleAdressenAsyncTests.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        var result = await client.GetPartijDigitaleAdressenAsync(partijUuid);
+
+        // Assert
+        Assert.NotNull(result);
+        Assert.Empty(result);
+    }
+
+    [Fact]
+    public async Task GetPartijDigitaleAdressenAsync_ReturnsEmptyListAndLogs_WhenPartijNotFound()
+    {
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Common.Test/Services/OpenKlantApi/GetPartijDigitaleAdressenAsyncTests.cs:110]: Addressed in d384f48 — added `_loggerMock.Verify(LogLevel.Warning, ...)` that asserts the warning contains the partijUuid.

File: InterneTaakAfhandeling.Common.Test/Services/OpenKlantApi/GetPartijDigitaleAdressenAsyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        Assert.Equal(2, result.Count);
+
+        Assert.Equal("jan@example.nl", result[0].Adres);
+        Assert.Equal("email", result[0].SoortDigitaalAdres);
+        Assert.True(result[0].IsStandaardAdres);
+
+        Assert.Equal("0612345678", result[1].Adres);
+        Assert.Equal("telefoonnummer", result[1].SoortDigitaalAdres);
+        Assert.False(result[1].IsStandaardAdres);
+    }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Common.Test/Services/OpenKlantApi/GetPartijDigitaleAdressenAsyncTests.cs:79]: Addressed in d384f48 — `MockHttpMessageHandler` now captures `LastRequest` and the happy-path test asserts the full URL including `?expand=digitaleAdressen`.
